lib/booking_system.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'reservation_creator'
+require 'pry'
+
+module Hotel
+  class BookingSystem
+    attr_reader :rooms, :reservations, :dates
+
+    def initialize
+      @reservations = []
+      @rooms = [*1..20]
+      @dates = []
+    end
+
+    def make_reservation(check_in_date:, check_out_date:)
+      # TODO find a room that's available
+
+      room_number = get_available_room(check_in_date: check_in_date, check_out_date: check_out_date)
+
+      reservation = ReservationCreator.new(check_in_date: check_in_date, check_out_date: check_out_date, room_number: room_number)
+
+
+
+      # @dates.each do |date_with_room|
+      #   date
+      #     if reservation.date_range.include?(date) && reservation.room_number == room
+      #
+      #       reservation.room_number += 1
+      #     end
+      #     binding.pry
+      #   end
+      # end
+
+
+      @reservations << reservation
+
+      return reservation
+    end
```

Comment:
I think from a birds-eye view, this method `make_reservation` is set up for success. It's outline is the following:

1. Figure out what the room number of the next available room is
1. Make a new instance of reservation with the following info: the passed in check in date, check out date, and the room number from step 1
1. add that reservation to the collection of reservations being recorded in this class on `@reservations`
1. return that reservation


I think that the rest of the work for wave 2 is just filling in the detail for those steps, and those details will be filled in with the helper methods. I think that this method looks really good as is though.